### ISSUE TITLE
Revoked certificates need exporter restart to clear metrics (sometimes)

### ISSUE THREAD

**Issue by wbollock** [aarnaud/vault-pki-exporter#34]
I've had trouble reproducing it, but have noticed in prod that revoked certificate metrics are not clearing on subsequent refresh intervals (even in multiple runs). The metrics continue to exist until the exporter is restarted. We deal with revoked certificates in 2 places:

* The actual map - https://github.com/aarnaud/vault-pki-exporter/blob/39ec19869c3357411ed2e1f1edd9b0b27885ba37/pkg/vault-mon/pki.go#L266
* And the Prometheus functions - https://github.com/aarnaud/vault-pki-exporter/blob/39ec19869c3357411ed2e1f1edd9b0b27885ba37/pkg/vault-mon/prometheus.go#L92

I'm not positive what the root cause is or how to reproduce this consistently

**Comment by aarnaud** [comment]
If I remember, `prom`  keep metrics, so even if you skip revoked certificate, the certificated that was previously in the list will stay.  So we need to clear the prom metrics. 
That explain why a restart solved your issue. 

Did I understood correctly ?

**Comment by aarnaud** [comment]
an because the first version of this exporter didn't handle CRL and new certificate replace previous one. I didn't need to clear prom metrics.  
now we need to delete the metrics or clear all before to loop on the cert list

**Comment by wbollock** [comment]
yes! managed to replicate the issue and a potential fix here https://github.com/aarnaud/vault-pki-exporter/pull/35
